### PR TITLE
Issue 9082 - beforeTest afterTest hooks not working with specify test interface in Mocha BDD

### PR DESCRIPTION
**Pull request by nagpalkaran95**
## Proposed changes
beforeTest & afterTest hooks in any service aren't working for `mocha` when test interface being used is `specify` instead of `it`. As per mocha [documentation](https://mochajs.org/#interfaces) for BDD, mocha allows usage of `it` and `specify` which is an alias for `it`.

Issue link: https://github.com/webdriverio/webdriverio/issues/9082
WebdriverIO v7 PR: https://github.com/webdriverio/webdriverio/pull/9083

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: packages/wdio-mocha-framework/src/constants.ts [hunk 1]
```diff
@@ -1,9 +1,15 @@
 export const INTERFACES = {
-    bdd: ['it', 'before', 'beforeEach', 'after', 'afterEach'],
+    bdd: ['it', 'specify', 'before', 'beforeEach', 'after', 'afterEach'],
     tdd: ['test', 'suiteSetup', 'setup', 'suiteTeardown', 'teardown'],
     qunit: ['test', 'before', 'beforeEach', 'after', 'afterEach']
 } as const
 
+export const TEST_INTERFACES = {
```
Comment (erwinheitzman) [inline packages/wdio-mocha-framework/src/constants.ts:7]: Why are these added? Having the INTERFACES seems like it already covers everything we need.

File: packages/wdio-mocha-framework/src/index.ts [hunk 2]
```diff
@@ -159,8 +159,7 @@ class MochaAdapter {
         }
 
         INTERFACES[type].forEach((fnName: string) => {
-            let testCommand = INTERFACES[type][0]
-            const isTest = [testCommand, testCommand + '.only'].includes(fnName)
+            const isTest = TEST_INTERFACES[type].flatMap((testCommand: string) => [testCommand, testCommand + '.only']).includes(fnName)
```
Comment (erwinheitzman) [inline packages/wdio-mocha-framework/src/index.ts:162]: This feel like an unnecessary change, you can use the existing setup with the addition of the alias, right?

**Review by erwinheitzman — CHANGES_REQUESTED** [review]
Thank you for this improvement, I left some feedback.

Could you take a look the linting issues and resolve them?

File: packages/wdio-mocha-framework/src/index.ts (hunk 2 above)
Comment (nagpalkaran95) [reply packages/wdio-mocha-framework/src/index.ts:162]: Having just INTERFACES constant doesn't solve this.
https://github.com/webdriverio/webdriverio/blob/main/packages/wdio-mocha-framework/src/index.ts#L162 - this statement makes just 'it' being taken as as a test interface.
I solved this in a similar way it is being done for jasmine. https://github.com/webdriverio/webdriverio/blob/main/packages/wdio-jasmine-framework/src/index.ts#L20 - it has interfaces and test interfaces are two different constants..

File: packages/wdio-mocha-framework/src/constants.ts (hunk 1 above)
Comment (nagpalkaran95) [reply packages/wdio-mocha-framework/src/constants.ts:7]: I solved this in a similar way it is being done for jasmine. https://github.com/webdriverio/webdriverio/blob/main/packages/wdio-jasmine-framework/src/index.ts#L20 - it has interfaces and test interfaces are two different constants..

File: packages/wdio-mocha-framework/src/constants.ts [hunk 3]
```diff
@@ -1,9 +1,15 @@
 export const INTERFACES = {
-    bdd: ['it', 'before', 'beforeEach', 'after', 'afterEach'],
+    bdd: ['it', 'specify', 'before', 'beforeEach', 'after', 'afterEach'],
```
Comment (erwinheitzman) [inline packages/wdio-mocha-framework/src/constants.ts:2]: I see that I forgot to mention that it's also missing `context` in this list, mind adding this as well so the list is complete?

Comment (nagpalkaran95) [reply packages/wdio-mocha-framework/src/constants.ts:2]: context is not a test interface. It is treated similar to describe. As describe isn't in this list, context shouldn't be a part of this list as well in my opinion..correct me if I am wrong though.

Comment (erwinheitzman) [reply packages/wdio-mocha-framework/src/constants.ts:2]: You're absolutely right. Gotta love switching context from work to an open source project hehe, sorry for overlooking that.

**Review by erwinheitzman — APPROVED** [review]
LGTM
